In [1]:
import pandas as pd
import numpy as np

files = {
    'expr.csv': lambda f: pd.read_csv(f, index_col=0, header=0).shape,
    'expr_2000.csv': lambda f: pd.read_csv(f, index_col=0, header=0).shape,
    'scPSD_embedding.txt.gz': lambda f: np.loadtxt(f).shape,
    'PCA_embedding.txt.gz': lambda f: np.loadtxt(f).shape,
    'scGNN_embedding.csv': lambda f: pd.read_csv(f).shape,

    'Fusion_sum-expr-scpsd-scgnn-pca.txt.gz': lambda f: np.loadtxt(f).shape,
    'Fusion_hadamard-expr-scpsd-scgnn-pca.txt.gz': lambda f: np.loadtxt(f).shape,
    'Fusion_attention-expr-scpsd-scgnn-pca.txt.gz': lambda f: np.loadtxt(f).shape,
    'Fusion_moe-expr-scpsd-scgnn-pca.txt.gz': lambda f: np.loadtxt(f).shape,
    'Fusion_residual-expr-scpsd-scgnn-pca.txt.gz': lambda f: np.loadtxt(f).shape,
    'Fusion_transformer-expr-scpsd-scgnn-pca.txt.gz': lambda f: np.loadtxt(f).shape,
}

In [2]:
print(f"{'File Name':<60} {'Shape':>15}")
print("-" * 75)

for fname, loader in files.items():
    try:
        shape = loader(fname)
        print(f"{fname:<60} {str(shape):>15}")
    except Exception as e:
        print(f"{fname:<60} {'Error'}")

File Name                                                              Shape
---------------------------------------------------------------------------
expr.csv                                                       (33694, 2998)
expr_2000.csv                                                   (2000, 2998)
scPSD_embedding.txt.gz                                          (2998, 2000)
PCA_embedding.txt.gz                                             (2998, 266)
scGNN_embedding.csv                                              (2998, 129)
Fusion_sum-expr-scpsd-scgnn-pca.txt.gz                          (2998, 2000)
Fusion_hadamard-expr-scpsd-scgnn-pca.txt.gz                     (2998, 2000)
Fusion_attention-expr-scpsd-scgnn-pca.txt.gz                    (2998, 2000)
Fusion_moe-expr-scpsd-scgnn-pca.txt.gz                           (2998, 128)
Fusion_residual-expr-scpsd-scgnn-pca.txt.gz                      (2998, 128)
Fusion_transformer-expr-scpsd-scgnn-pca.txt.gz                   (2998, 128)


In [3]:
pd.read_csv('cluster.csv', header=None)

0
0          Macrophage
1              T cell
2          Macrophage
3     Epithelial cell
4           Mast cell
...               ...
2993  Epithelial cell
2994           T cell
2995           T cell
2996       Macrophage
2997        Mast cell

[2998 rows x 1 columns]

In [4]:
pd.read_csv('cluster.csv', header=None)[0].value_counts()

0
T cell              888
Epithelial cell     790
Macrophage          666
B cell              276
Fibroblast          185
Endothelial cell    112
Mast cell            81
Name: count, dtype: int64

### Feature Embedding Files and Their Dimensions

| Feature ID | File Name                                                               | Dimension      |
|------------|-------------------------------------------------------------------------|----------------|
| Feature0   | expr.csv                                                                | (33694, 2998)  |
| Feature1   | expr_2000.csv                                                           | (2000, 2998)   |
| Feature2   | scPSD_embedding.txt.gz                                                  | (2998, 2000)   |
| Feature3   | PCA_embedding.txt.gz                                                    | (2998, 266)    |
| Feature4   | scGNN_embedding.csv                                                     | (2998, 129)    |
| Feature5   | Fusion_sum-expr-scpsd-scgnn-pca.txt.gz                                  | (2998, 2000)   |
| Feature6   | Fusion_hadamard-expr-scpsd-scgnn-pca.txt.gz                             | (2998, 2000)   |
| Feature7   | Fusion_attention-expr-scpsd-scgnn-pca.txt.gz                            | (2998, 2000)   |
| Feature8   | Fusion_moe-expr-scpsd-scgnn-pca.txt.gz                                  | (2998, 128)    |
| Feature9   | Fusion_residual-expr-scpsd-scgnn-pca.txt.gz                             | (2998, 128)    |
| Feature10  | Fusion_transformer-expr-scpsd-scgnn-pca.txt.gz                          | (2998, 128)    |

